diff --git a/nginx.spec.org b/nginx.spec
index fdf002d..3669507 100644
--- a/nginx.spec.org
+++ b/nginx.spec
@@ -48,7 +48,7 @@ BuildRequires: systemd
 # end of distribution specific definitions
 
 %define main_version 1.10.3
-%define main_release 1%{?dist}.ngx
+%define main_release dda1%{?dist}.ngx
 
 %define bdir %{_builddir}/%{name}-%{main_version}
 
@@ -61,6 +61,7 @@ Summary: High performance web server
 Name: nginx
 Version: %{main_version}
 Release: %{main_release}
+Packager: Diogo Dutra <dutradda@gmail.com>
 Vendor: Nginx, Inc.
 URL: http://nginx.org/
 Group: %{_group}
@@ -78,6 +79,7 @@ Source10: nginx.suse.logrotate
 Source11: nginx-debug.service
 Source12: COPYRIGHT
 Source13: nginx.check-reload.sh
+Source14: headers-more-nginx-module-0.29.tar.gz
 
 License: 2-clause BSD-like license
 
@@ -96,7 +98,7 @@ a mail proxy server.
 %endif
 
 %prep
-%setup -q
+%setup -b 14 -q
 cp %{SOURCE2} .
 sed -e 's|%%DEFAULTSTART%%|2 3 4 5|g' -e 's|%%DEFAULTSTOP%%|0 1 6|g' \
     -e 's|%%PROVIDES%%|nginx|g' < %{SOURCE2} > nginx.init
@@ -107,13 +109,41 @@ sed -e 's|%%DEFAULTSTART%%||g' -e 's|%%DEFAULTSTOP%%|0 1 2 3 4 5 6|g' \
 ./configure %{BASE_CONFIGURE_ARGS} \
     --with-cc-opt="%{WITH_CC_OPT}" \
     --with-ld-opt="%{WITH_LD_OPT}" \
+    --with-pcre \
+    --without-http_autoindex_module \
+    --without-http_empty_gif_module \
+    --without-http_geo_module \
+    --without-http_scgi_module \
+    --without-http_ssi_module \
+    --without-http_split_clients_module \
+    --without-http_upstream_ip_hash_module \
+    --without-http_userid_module \
+    --without-http_uwsgi_module \
+    --without-mail_imap_module \
+    --without-mail_pop3_module \
+    --without-mail_smtp_module \
+    --add-module=../headers-more-nginx-module-0.29 \
     --with-debug
 make %{?_smp_mflags}
 %{__mv} %{bdir}/objs/nginx \
     %{bdir}/objs/nginx-debug
 ./configure %{BASE_CONFIGURE_ARGS} \
     --with-cc-opt="%{WITH_CC_OPT}" \
-    --with-ld-opt="%{WITH_LD_OPT}"
+    --with-ld-opt="%{WITH_LD_OPT}" \
+    --with-pcre \
+    --without-http_autoindex_module \
+    --without-http_empty_gif_module \
+    --without-http_geo_module \
+    --without-http_scgi_module \
+    --without-http_ssi_module \
+    --without-http_split_clients_module \
+    --without-http_upstream_ip_hash_module \
+    --without-http_userid_module \
+    --without-http_uwsgi_module \
+    --without-mail_imap_module \
+    --without-mail_pop3_module \
+    --without-mail_smtp_module \
+    --add-module=../headers-more-nginx-module-0.29
 make %{?_smp_mflags}
 
 %install
@@ -121,10 +151,13 @@ make %{?_smp_mflags}
 %{__make} DESTDIR=$RPM_BUILD_ROOT INSTALLDIRS=vendor install
 
 %{__mkdir} -p $RPM_BUILD_ROOT%{_datadir}/nginx
-%{__mv} $RPM_BUILD_ROOT%{_sysconfdir}/nginx/html $RPM_BUILD_ROOT%{_datadir}/nginx/
 
 %{__rm} -f $RPM_BUILD_ROOT%{_sysconfdir}/nginx/*.default
 %{__rm} -f $RPM_BUILD_ROOT%{_sysconfdir}/nginx/fastcgi.conf
+%{__rm} -rf $RPM_BUILD_ROOT%{_sysconfdir}/nginx/html
+%{__rm} -rf $RPM_BUILD_ROOT%{_sysconfdir}/nginx/scgi_params
+%{__rm} -rf $RPM_BUILD_ROOT%{_sysconfdir}/nginx/uwsgi_params
+%{__rm} -rf $RPM_BUILD_ROOT%{_sysconfdir}/nginx/koi-*
 
 %{__mkdir} -p $RPM_BUILD_ROOT%{_localstatedir}/log/nginx
 %{__mkdir} -p $RPM_BUILD_ROOT%{_localstatedir}/run/nginx
@@ -139,11 +172,10 @@ cd $RPM_BUILD_ROOT%{_sysconfdir}/nginx && \
     $RPM_BUILD_ROOT%{_datadir}/doc/%{name}-%{main_version}/
 
 %{__mkdir} -p $RPM_BUILD_ROOT%{_sysconfdir}/nginx/conf.d
+%{__mkdir} -p $RPM_BUILD_ROOT%{_sysconfdir}/nginx/vhosts
 %{__rm} $RPM_BUILD_ROOT%{_sysconfdir}/nginx/nginx.conf
 %{__install} -m 644 -p %{SOURCE4} \
     $RPM_BUILD_ROOT%{_sysconfdir}/nginx/nginx.conf
-%{__install} -m 644 -p %{SOURCE5} \
-    $RPM_BUILD_ROOT%{_sysconfdir}/nginx/conf.d/default.conf
 
 %{__mkdir} -p $RPM_BUILD_ROOT%{_sysconfdir}/sysconfig
 %{__install} -m 644 -p %{SOURCE3} \
@@ -197,16 +229,12 @@ cd $RPM_BUILD_ROOT%{_sysconfdir}/nginx && \
 
 %dir %{_sysconfdir}/nginx
 %dir %{_sysconfdir}/nginx/conf.d
+%dir %{_sysconfdir}/nginx/vhosts
 %{_sysconfdir}/nginx/modules
 
 %config(noreplace) %{_sysconfdir}/nginx/nginx.conf
-%config(noreplace) %{_sysconfdir}/nginx/conf.d/default.conf
 %config(noreplace) %{_sysconfdir}/nginx/mime.types
 %config(noreplace) %{_sysconfdir}/nginx/fastcgi_params
-%config(noreplace) %{_sysconfdir}/nginx/scgi_params
-%config(noreplace) %{_sysconfdir}/nginx/uwsgi_params
-%config(noreplace) %{_sysconfdir}/nginx/koi-utf
-%config(noreplace) %{_sysconfdir}/nginx/koi-win
 %config(noreplace) %{_sysconfdir}/nginx/win-utf
 
 %config(noreplace) %{_sysconfdir}/logrotate.d/nginx
@@ -224,9 +252,6 @@ cd $RPM_BUILD_ROOT%{_sysconfdir}/nginx && \
 
 %attr(0755,root,root) %dir %{_libdir}/nginx
 %attr(0755,root,root) %dir %{_libdir}/nginx/modules
-%dir %{_datadir}/nginx
-%dir %{_datadir}/nginx/html
-%{_datadir}/nginx/html/*
 
 %attr(0755,root,root) %dir %{_localstatedir}/cache/nginx
 %attr(0755,root,root) %dir %{_localstatedir}/log/nginx
@@ -308,6 +333,9 @@ if [ $1 -ge 1 ]; then
 fi
 
 %changelog
+* Tue Mar 21 2017 Diogo Dutra <dutradda@gmail.com>
+- 1.10.3 with Headers More module
+
 * Tue Jan 31 2017 Konstantin Pavlov <thresh@nginx.com>
 - 1.10.3
 - Extended hardening build flags.
