--- nginx.spec.org	2016-08-01 16:09:52.000000000 +0900
+++ nginx.spec	2016-08-01 16:21:02.000000000 +0900
@@ -61,7 +61,7 @@
 # end of distribution specific definitions
 
 %define main_version                 1.10.1
-%define main_release                 1%{?dist}.ngx
+%define main_release                 ff1%{?dist}.ngx
 %define module_xslt_version          %{main_version}
 %define module_xslt_release          1%{?dist}.ngx
 %define module_geoip_version         %{main_version}
@@ -96,17 +96,8 @@
         --group=%{nginx_group} \
         --with-http_ssl_module \
         --with-http_realip_module \
-        --with-http_addition_module \
-        --with-http_sub_module \
-        --with-http_dav_module \
-        --with-http_flv_module \
-        --with-http_mp4_module \
-        --with-http_gunzip_module \
         --with-http_gzip_static_module \
-        --with-http_random_index_module \
-        --with-http_secure_link_module \
         --with-http_stub_status_module \
-        --with-http_auth_request_module \
         --with-http_xslt_module=dynamic \
         --with-http_image_filter_module=dynamic \
         --with-http_geoip_module=dynamic \
@@ -116,16 +107,26 @@
         --with-stream \
         --with-stream_ssl_module \
         --with-http_slice_module \
-        --with-mail \
-        --with-mail_ssl_module \
         --with-file-aio \
-        --with-ipv6 \
-        %{?with_http2:--with-http_v2_module}")
+        --with-pcre \
+        --without-http_autoindex_module \
+        --without-http_empty_gif_module \
+        --without-http_geo_module \
+        --without-http_scgi_module \
+        --without-http_ssi_module \
+        --without-http_split_clients_module \
+        --without-http_upstream_ip_hash_module \
+        --without-http_userid_module \
+        --without-http_uwsgi_module \
+        --without-mail_imap_module \
+        --without-mail_pop3_module \
+        --without-mail_smtp_module")
 
 Summary: High performance web server
 Name: nginx
 Version: %{main_version}
 Release: %{main_release}
+Packager: feedforce Inc. <socialplus_admin@feedforce.jp>
 Vendor: Nginx, Inc.
 URL: http://nginx.org/
 Group: %{_group}
@@ -143,6 +144,7 @@
 Source11: nginx-debug.service
 Source12: COPYRIGHT
 Source13: njs-%{module_njs_shaid}.tar.gz
+Source14: headers-more-nginx-module-0.30.tar.gz
 
 License: 2-clause BSD-like license
 
@@ -210,6 +212,7 @@
 
 %prep
 %setup -q
+%setup -b 14 -q
 tar xvzf %SOURCE13
 cp %{SOURCE2} .
 sed -e 's|%%DEFAULTSTART%%|2 3 4 5|g' -e 's|%%DEFAULTSTOP%%|0 1 6|g' \
@@ -219,6 +222,7 @@
 
 %build
 ./configure %{COMMON_CONFIGURE_ARGS} \
+    --add-module=../headers-more-nginx-module-0.29 \
     --with-cc-opt="%{WITH_CC_OPT}" \
     --with-debug
 make %{?_smp_mflags}
@@ -250,10 +254,13 @@
 find %{buildroot} -type f -name nginx.so -exec chmod u+w '{}' \;
 
 %{__mkdir} -p $RPM_BUILD_ROOT%{_datadir}/nginx
-%{__mv} $RPM_BUILD_ROOT%{_sysconfdir}/nginx/html $RPM_BUILD_ROOT%{_datadir}/nginx/
 
 %{__rm} -f $RPM_BUILD_ROOT%{_sysconfdir}/nginx/*.default
 %{__rm} -f $RPM_BUILD_ROOT%{_sysconfdir}/nginx/fastcgi.conf
+%{__rm} -rf $RPM_BUILD_ROOT%{_sysconfdir}/nginx/html
+%{__rm} -rf $RPM_BUILD_ROOT%{_sysconfdir}/nginx/scgi_params
+%{__rm} -rf $RPM_BUILD_ROOT%{_sysconfdir}/nginx/uwsgi_params
+%{__rm} -rf $RPM_BUILD_ROOT%{_sysconfdir}/nginx/koi-*
 
 %{__mkdir} -p $RPM_BUILD_ROOT%{_localstatedir}/log/nginx
 %{__mkdir} -p $RPM_BUILD_ROOT%{_localstatedir}/run/nginx
@@ -268,11 +275,10 @@
     $RPM_BUILD_ROOT%{_datadir}/doc/%{name}-%{main_version}/
 
 %{__mkdir} -p $RPM_BUILD_ROOT%{_sysconfdir}/nginx/conf.d
+%{__mkdir} -p $RPM_BUILD_ROOT%{_sysconfdir}/nginx/vhosts
 %{__rm} $RPM_BUILD_ROOT%{_sysconfdir}/nginx/nginx.conf
 %{__install} -m 644 -p %{SOURCE4} \
     $RPM_BUILD_ROOT%{_sysconfdir}/nginx/nginx.conf
-%{__install} -m 644 -p %{SOURCE5} \
-    $RPM_BUILD_ROOT%{_sysconfdir}/nginx/conf.d/default.conf
 
 %{__mkdir} -p $RPM_BUILD_ROOT%{_sysconfdir}/sysconfig
 %{__install} -m 644 -p %{SOURCE3} \
@@ -335,16 +341,12 @@
 
 %dir %{_sysconfdir}/nginx
 %dir %{_sysconfdir}/nginx/conf.d
+%dir %{_sysconfdir}/nginx/vhosts
 %{_sysconfdir}/nginx/modules
 
 %config(noreplace) %{_sysconfdir}/nginx/nginx.conf
-%config(noreplace) %{_sysconfdir}/nginx/conf.d/default.conf
 %config(noreplace) %{_sysconfdir}/nginx/mime.types
 %config(noreplace) %{_sysconfdir}/nginx/fastcgi_params
-%config(noreplace) %{_sysconfdir}/nginx/scgi_params
-%config(noreplace) %{_sysconfdir}/nginx/uwsgi_params
-%config(noreplace) %{_sysconfdir}/nginx/koi-utf
-%config(noreplace) %{_sysconfdir}/nginx/koi-win
 %config(noreplace) %{_sysconfdir}/nginx/win-utf
 
 %config(noreplace) %{_sysconfdir}/logrotate.d/nginx
@@ -362,9 +364,6 @@
 
 %attr(0755,root,root) %dir %{_libdir}/nginx
 %attr(0755,root,root) %dir %{_libdir}/nginx/modules
-%dir %{_datadir}/nginx
-%dir %{_datadir}/nginx/html
-%{_datadir}/nginx/html/*
 
 %attr(0755,root,root) %dir %{_localstatedir}/cache/nginx
 %attr(0755,root,root) %dir %{_localstatedir}/log/nginx
@@ -560,6 +559,9 @@
 fi
 
 %changelog
+* Mon Aug 01 2016 Takashi Masuda <masutaka@feedforce.jp>
+- 1.10.1 with Headers More module
+
 * Tue May 31 2016 Konstantin Pavlov <thresh@nginx.com>
 - 1.10.1
 
